packages/@aws-cdk/aws-kms/lib/key.ts

```diff
@@ -365,12 +365,26 @@ export class Key extends KeyBase {
   constructor(scope: Construct, id: string, props: KeyProps = {}) {
     super(scope, id);
 
-    this.policy = props.policy || new iam.PolicyDocument();
-    this.trustAccountIdentities = props.trustAccountIdentities || false;
-    if (this.trustAccountIdentities) {
-      this.allowAccountIdentitiesToControl();
+    const defaultKeyPoliciesEnabled = FeatureFlags.of(this).isEnabled(cxapi.KMS_DEFAULT_KEY_POLICIES);
+
+    this.policy = props.policy ?? new iam.PolicyDocument();
+    if (defaultKeyPoliciesEnabled) {
+      if (props.trustAccountIdentities === false) {
+        Annotations.of(this).addWarning('`trustAccountIdentities` has no impact if the @aws-cdk/aws-kms:defaultKeyPolicies feature flag is set');
```

Comment:
I wonder - should we throw an exception here instead? Is a warning a little too soft?

Reply:
I would also be okay with throwing. A nonsensical combination of inputs is usually okay for an exception in my book.